src/measure_theory/measure/lebesgue.lean

```diff
@@ -239,11 +261,150 @@ lemma map_volume_mul_right {a : ℝ} (h : a ≠ 0) :
   measure.map (* a) volume = ennreal.of_real (abs a⁻¹) • volume :=
 by simpa only [mul_comm] using real.map_volume_mul_left h
 
+@[simp] lemma volume_preimage_mul_right {a : ℝ} (h : a ≠ 0) (s : set ℝ) :
+  volume ((* a) ⁻¹' s) = ennreal.of_real (abs a⁻¹) * volume s :=
+calc volume ((* a) ⁻¹' s) = measure.map (* a) volume s :
+  ((homeomorph.mul_right' a h).to_measurable_equiv.map_apply s).symm
+... = ennreal.of_real (abs a⁻¹) * volume s : by { rw map_volume_mul_right h, refl }
+
 @[simp] lemma map_volume_neg : measure.map has_neg.neg (volume : measure ℝ) = volume :=
 eq.symm $ real.measure_ext_Ioo_rat $ λ p q,
   by simp [show measure.map has_neg.neg volume (Ioo (p : ℝ) q) = _,
     from measure.map_apply measurable_neg measurable_set_Ioo]
 
+/-!
+### Images of the Lebesgue measure under translation/linear maps in ℝⁿ
+-/
+
+lemma map_volume_pi_add_left (a : ι → ℝ) : measure.map ((+) a) volume = volume :=
+begin
+  refine (measure.pi_eq (λ s hs, _)).symm,
+  have A : has_add.add a ⁻¹' (set.pi univ (λ (i : ι), s i))
+    = set.pi univ (λ (i : ι), ((+) (a i)) ⁻¹' (s i)), by { ext, simp },
+  rw [measure.map_apply (measurable_const_add a) (measurable_set.univ_pi_fintype hs), A,
+      volume_pi_pi],
+  { simp only [volume_preimage_add_left] },
+  { exact λ i, measurable_const_add (a i) (hs i) }
+end
+
+@[simp] lemma volume_pi_preimage_add_left (a : ι → ℝ) (s : set (ι → ℝ)) :
+  volume (((+) a) ⁻¹' s) = volume s :=
+calc volume (((+) a) ⁻¹' s) = measure.map ((+) a) volume s :
+  ((homeomorph.add_left a).to_measurable_equiv.map_apply s).symm
+... = volume s : by rw map_volume_pi_add_left
+
+open matrix
+
+/-- A diagonal matrix rescales Lebesgue according to its determinant. -/
+lemma smul_map_diagonal_volume_pi [decidable_eq ι] {D : ι → ℝ} (h : det (diagonal D) ≠ 0) :
+  ennreal.of_real (abs (det (diagonal D))) • measure.map ((diagonal D).to_lin') volume = volume :=
+begin
```

Comment:
Is it worth stating this in terms of `matrix.is_diag`? It seems odd to talk about `det (diagonal D)` instead of just the product of `D`.

Reply:
This is just an intermediate step to prove the same result for all matrices, in the form it is needed for the forthcoming proof. So I'd rather keep it as is, but I have clarified this in the docstring.